crates/bevy_reflect/src/array.rs

```diff
@@ -374,7 +374,7 @@ impl<'a> Iterator for ArrayIter<'a> {
     #[inline]
     fn next(&mut self) -> Option<Self::Item> {
         let value = self.array.get(self.index);
-        self.index += 1;
+        self.index += value.is_some() as usize;
```

Comment:
Awesome! Thanks for adding these as well!

Would also be able to duplicate that test for each of these changes as well?

Reply:
I couldnt test the overflow for most of them because I couldnt create an iterator with enough elements.
But the logic of not incrementing past the first None I can test, so I've added tests for that